draft-ietf-masque-connect-udp-listen.md

```diff
@@ -153,11 +170,103 @@ UDP Payload:
 : The unmodified UDP Payload of this proxied UDP packet (referred to as "data
 octets" in {{UDP}}).
 
+# Context ID {#contextid}
+
+Context ID values MUST always be greater than 0. A payload with value 0 MUST
+be dropped by the receiving party.
+
```

Comment:
I've put comments inline for this paragraph, but here's an attempt at tweaking the paragraph to explain the concepts without normative 2119 verbs - those come later:

> This extension leverages context IDs (see Section 4 of CONNECT-UDP) to compress the target IP address and port when encoding datagrams on the wire. Either endpoint can register a context ID and the IP/ports it's associated with by sending a COMPRESSION_ASSIGN capsule to its peer. The peer will then echo that capsule to indicate it's received it. From then on, both endpoints are aware of the context ID and can send compressed datagrams. Later, any endpoint can decide to close the compression context by sending a COMPRESSION_CLOSE capsule.